app/models/policy_host.rb

```diff
@@ -8,10 +8,26 @@ class PolicyHost < ApplicationRecord
   validates :policy, presence: true
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
+  validate :host_supported?, on: :create
+
+  scope :supported_os_versions, lambda { |policy_id, host_os_major_version|
+    profile = Profile.find_by(policy_id: policy_id)
+    if host_os_major_version.to_s == profile&.os_major_version.to_s
+      profile&.supported_minor_versions
+    end
+  }
 
   def self.import_from_policy(policy_id, host_ids)
     import(host_ids.map do |host_id|
       { host_id: host_id, policy_id: policy_id }
     end)
   end
+
+  def host_supported?
+    if PolicyHost.supported_os_versions(policy&.id, host&.os_major_version).include?(host&.os_minor_version.to_s)
+      return
+    end
+
+    errors.add(:host, 'os version is unsupported for this policy')
```

Comment:
Using a scope here is a bit of an antipattern, especially in the same exact class. My oversimplified idea would be something like:

```suggestion
    errors.add(:host, 'Unsupported OS major version') unless host.os_major_version == initial_profile.os_major_version
    
    errors.add(:host, 'Unsupported OS minor version') if SupportedSsg.ssg_versions_for_os(host.os_major_version, host.os_minor_version).include?(supported_benchmark_versions)
```

Reply:
Yep, I like this idea! Thanks